spec/Section 5 -- Validation.md

```diff
@@ -1304,15 +1304,26 @@ fragment resourceFragment on Resource {
 
 **Formal Specification**
 
-- For each input Value {value} in the document:
+- For each literal Input Value {value} in the document:
```

Comment:
Suggestion to makes the locations explicit:

```suggestion
- For each argument Value and variable DefaultValue {value} in the document:
```

I think this covers it all? 

It makes it easier for implementers to make sure they cover everything and removes ambiguity about list/object literals (they are a `Value` that may contain nested `Value`, we only need to validate the root value)

The algorithm could rule out variables as the first step:

```
  - If {value} is not a variable
    - ...
```

Reply:
If we're trying to cover everything should we indicate that the arguments don't just include field arguments, but also directive arguments? And in future, we'd also be including fragment arguments (which I can't remember if we decided to call something other than "arguments" in the end?).

```suggestion
For each literal field argument value, literal directive argument value, literal
fragment argument value and variable default value {value} in the document:
```

I think this is getting a bit of a mouthful though, and puts more burden on spec updates. Let's stick with "input value", but maybe define it in a follow-up PR?

Reply:
If we read `argument` in the "grammar" acceptation, the nice thing with `argument` is that it works for both fields and directive, they're using the same grammar rule. It'd actually be cool to have "AST coordinates" that we could refer to to easily identify those locations.

Agree that listing everything exhaustively is taking the risk this content gets up to date 👍

Reply:
In all cases, LGTM, resolving that conversation.